tests/tests.rs

```diff
@@ -2,111 +2,137 @@ use enum_assoc::Assoc;
 
 // A bit of mock data
 const WA: &'static str = "wa";
-fn some_str_func(s: &'static str) -> String
-{
+fn some_str_func(s: &'static str) -> String {
     String::from("I was created in a function") + s
 }
 
 #[derive(Assoc)]
 #[func(const fn foo(&self) -> u8)]
 #[func(pub fn bar(&self) -> &'static str)]
 #[func(pub fn maybe_foo(&self) -> Option<u8>)]
-enum TestEnum
-{
-    #[assoc(foo = 255)] 
-    #[assoc(bar = "wow")] 
+enum TestEnum {
+    #[assoc(foo = 255)]
+    #[assoc(bar = "wow")]
     Variant1,
-    #[assoc(foo = 1 + 7)] 
-    #[assoc(bar = "wee")] 
+    #[assoc(foo = 1 + 7)]
+    #[assoc(bar = "wee")]
     Variant2,
     #[assoc(foo = 0)]
-    #[assoc(bar = WA)] 
-    #[assoc(maybe_foo = 18 + 2)] 
-    Variant3
+    #[assoc(bar = WA)]
+    #[assoc(maybe_foo = 18 + 2)]
+    Variant3,
 }
 
 #[derive(Assoc)]
 #[func(pub fn foo(&self, param: u8) -> Option<u8>)]
 #[func(pub fn bar(&self, param: &'static str) -> String)]
 #[func(pub fn baz<T: std::fmt::Debug>(&self, param: T) -> Option<String>)]
-enum TestEnum2
-{
-    #[assoc(bar = String::new() + param)] 
+enum TestEnum2 {
+    #[assoc(bar = String::new() + param)]
     Variant1,
-    #[assoc(foo = 12 + param)] 
-    #[assoc(bar = String::from("Hello") + param)] 
+    #[assoc(foo = 12 + param)]
+    #[assoc(bar = String::from("Hello") + param)]
+    Variant2,
+    #[assoc(bar = some_str_func("!"))]
+    #[assoc(baz = format!("{:?}", param))]
+    Variant3,
+}
+
+#[derive(Assoc)]
+#[func(pub const fn bar(&self) -> &'static str)]
+enum TestEnum3 {
```

Comment:
This name is already used for the enum in the module. Because of the module it is valid to let them both have the same name, but it might be a bit less confusing to use a different name